server/src/Application/Validation/ValidationHandler.cs

```diff
@@ -22,4 +21,15 @@ public void Validate<T>(T instance)
 
         validator.ValidateAndThrow(instance);
     }
+
+    public async Task ValidateAsync<T>(T instance)
```

Comment:
Seems like duplicate logic, could make a private function which gets the validator